### ISSUE TITLE
API: define __copy__ NDFrame

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#8571]
from [SO](http://stackoverflow.com/questions/26421221/propogation-of-metadata-from-dataframe-to-series/26423596#26423596)

this allows `copy.copy(df)` to work properly

by definition 'always' deep

```
__copy__ = copy
__deepcopy__ = copy
```

**Comment by gliptak** [comment]
@jreback Was this closed by https://github.com/jreback/pandas/commit/d82196c2932acc63401edbbafd2af24df11585a4 (from above)? Thanks

**Comment by jreback** [comment]
no that's something else

**Comment by gliptak** [comment]
"closes #8571 by defining __copy__/__deepcopy__"?

**Comment by jreback** [comment]
it was not merged

**Comment by gliptak** [comment]
I see. So would adding ```__copy__``/```__deepcopy__``` as in http://stackoverflow.com/a/15774013/304690 be appropriate?

**Comment by jreback** [comment]
I think you can just define them as I did at the top

**Comment by gliptak** [comment]
```
import pandas as pd
import numpy as np
import copy
pd.Series._metadata.append('units')
df = pd.DataFrame(data=np.random.randn(5,2), columns=['A', 'B'])
df['A'].units = 's'
df['B'].units = 'm'
s = copy.copy(df['A'])
print(s.units)
```
prints
```
s
```
on
```
INSTALLED VERSIONS
------------------
commit: 504ad4641e5b3a87b655da5dd31840de12495e97
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-16-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0+105.g504ad46.dirty
nose: 1.3.7
pip: 8.1.1
setuptools: 20.2.2
Cython: None
numpy: 1.10.4
scipy: 0.17.0
statsmodels: None
xarray: None
IPython: 4.1.2
sphinx: None
patsy: None
dateutil: 2.5.1
pytz: 2016.2
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None
```

**Comment by jorisvandenbossche** [comment]
@jreback I think this was closed by #15444?

**Comment by jreback** [comment]
yes
